### PR TITLE
Add withAzureCredentials step

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Add new step to interact with azure

## Why is it important?

Facilitate the consumer's pipelines.

## Related issues
Closes #ISSUE

### CODE REVIEW

File: vars/README.md [hunk 1]
```diff
@@ -2491,6 +2491,23 @@ withAPM(serviceName: 'apm-traces', transactionNAme: 'test') {
...
+}
+
+withAzureCredentials(path: '/foo', credentialsFile: '.credentials.json') {
+  // block
+}
+```
+
+* path: root folder where the credentials file will be stored. (Optional). Default: ${HOME} env variable
+* credentialsFile: name of the file with the credentials. (Optional). Default: .credentials.json
+* secret: Name of the secret on the the vault root path. (Optional). Default: 'secret/apm-team/ci/apm-agent-dotnet-azures'
```
Comment (kuisathaverat) [inline vars/README.md:2509]:
```suggestion
* secret: Name of the secret on the the vault root path. (Optional). Default: 'secret/apm-team/ci/apm-agent-dotnet-azure'
```

**Review by kuisathaverat — APPROVED** [review]

File: vars/withAzureCredentials.groovy [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  }
+  def client_id = props?.data?.client_id
+  def client_secret = props?.data?.client_secret
+  def subscription_id = props?.data?.subscription_id
+  def tenant_id = props?.data?.tenant_id
+  if (client_id == null || client_secret == null || subscription_id == null || tenant_id == null) {
+    error 'withAzureCredentials: was not possible to get authentication info'
+  }
+  dir(path) {
+    writeFile file: credentialsFile, text: """
```
Comment (kuisathaverat) [inline vars/withAzureCredentials.groovy:42]: can we set the permissions of the file to 0600?

File: vars/withAzureCredentials.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+withAzureCredentials(path: '/foo', credentialsFile: '.credentials.json') {
+  // block
+}
+```
+
+* path: root folder where the credentials file will be stored. (Optional). Default: ${HOME} env variable
+* credentialsFile: name of the file with the credentials. (Optional). Default: .credentials.json
+* secret: Name of the secret on the the vault root path. (Optional). Default: 'secret/apm-team/ci/apm-agent-dotnet-azures'
```
Comment (v1v) [inline vars/withAzureCredentials.txt:15]:
```suggestion
* secret: Name of the secret on the the vault root path. (Optional). Default: 'secret/apm-team/ci/apm-agent-dotnet-azure'
```

File: vars/withAzureCredentials.groovy (hunk 2 above)
Comment (v1v) [reply vars/withAzureCredentials.groovy:42]: `writeFile` does not support permissions yet (https://issues.jenkins.io/browse/JENKINS-47747)  I'm a bit reluctant to add a sh/bat step only for that. Do you see any issues with the existing approach?

**Pull request merged**